common/remote/sockfile.cpp

```diff
@@ -6314,12 +6407,28 @@ class CRemoteFileServer : implements IRemoteFileServer, public CInterface
             MemoryBufferBuilder outBuilder(resultBuffer, out->getMinRecordSize());
             if (outFmt_Binary == outputFormat)
             {
-                DelayedSizeMarker dataLenMarker(reply); // data length
+                if (compressor)
+                {
+                    compressMb.setEndian(__BIG_ENDIAN);
+                    compressMb.append(reply);
+                }
+
+                DelayedMarker<size32_t> dataLenMarker(compressor ? compressMb : reply); // data length
+
+                if (compressor)
+                {
+                    size32_t initialSz = replyLimit >= 0x10000 ? 0x10000 : replyLimit;
```

Comment:
Should the initialSz be rounded up to a integer multiple of out->getMinRecordSize() ?

Reply:
won't help, as the size here (to ICompressor->open) is the initial compressed output buffer size.
So won't be a multiple of inputs size (even if were fixed size record widths)